src/NServiceBus.Hosting.Windows/StartableAndStoppable/EndpointConfigurationExtensions.cs

```diff
@@ -0,0 +1,27 @@
+namespace NServiceBus
+{
+    using System.Collections.Generic;
+    using Configuration.AdvanceExtensibility;
+
+    /// <summary>
+    /// Extension methods for EndpointConfiguration
+    /// </summary>
+    public static class EndpointConfigurationExtensions
+    {
+        /// <summary>
+        /// Register a specific instance of an IWantToRunWhenEndpointStartsAndStops implementation
+        /// </summary>
+        public static void RunWhenEndpointStartsAndStops(this EndpointConfiguration configuration, IWantToRunWhenEndpointStartsAndStops callbackObject)
+        {
+            var settings = configuration.GetSettings();
+
```

Comment:
I think the code would be more clean if you do a `TryGet` instead? (and only do a `Set` if it wasn't found)